### ISSUE TITLE
Incorrect warnings (Rust)

### ISSUE THREAD

**Issue by DavidAntliff** [chylex/IntelliJ-Inspection-Lens#28]
Using version 1.4.1 with RustRover 2024.4 EAP (243.20847.53).

Mostly working well, but occasionally gets confused:

![image](https://github.com/user-attachments/assets/b1292e8d-1e0e-4f16-b393-42211726026a)

These two spurious warnings only appear if the plugin is enabled.

Enabled:
![image](https://github.com/user-attachments/assets/16129cf4-97aa-4337-ba9f-b9933ba4dcd7)

Disabled:
![image](https://github.com/user-attachments/assets/9cf4c074-4b91-4ef6-a238-717fd4bba478)

To be clear, those two extra warnings only appear if the plugin is **enabled**.

Incidentally, there is no variable called `physical_address` in this file, but it does occur as a struct _field_ in a different file.

And I'm not sure what's going on with that `does not need to be mutable` warning - there is no mutable variable there.

Let me know if there's more info I can provide.

**Comment by chylex** [comment]
Does it happen consistently, i.e. if you re-enable the plugin do the spurious warnings appear again even after restarting RustRover?

**Comment by DavidAntliff** [comment]
Yes, it seems persistent across IDE restarts, plugin disable/re-enable, as well as invalidate caches. It’s almost like it’s picking up warnings from other files and displaying them for the wrong file…

**Comment by chylex** [comment]
Are the project sources available?

**Comment by DavidAntliff** [comment]
> Are the project sources available?

Unfortunately not with this project, but I'll keep an eye out for anything similar with any open-source projects I work on, and report back. If you'd prefer to close this issue now, that's cool - I can re-open it once I find another example.

(Note that I cleaned up all warnings from all files and these spurious warnings also disappeared, but I can't tell you when exactly that happened, sadly).
